src/docs/theia_ai.md

```diff
@@ -393,6 +394,53 @@ For further details, we recommend reviewing the available LLM provider in Theia
 
 Please note that Theia AI currently does not provide a fixed contribution point for Language Models, yet. This is due to the fact that we are working on supporting more models and also capabilities of new LLMs are emerging at the moment, such as function calling and structured output. We plan to consolidate the LLM Provider interfaces within the next months while adding more LLM Providers to the core framework. We are happy for feedback and contributions in this area.
 
+## Changesets
+
+ChangeSets in Theia AI provide a mechanism for AI agents (and therefore the underlying LLMs) to propose changes to users. These proposed changes can then be reviewed, accepted, refined, or declined by the user. Theia AI offers framework support for generic ChangeSets, a default UI integrated in the generic, reusable Chat and Theia AI includes a default implementation for file-based changes. This default implementation is utilized in the Theia IDE, particularly with the [Theia Coder agent](/docs/theia_coder). However, adopters can provide alternative implementations to handle different types of changes, such as modifications to databases, structured models, or other domain-specific data.
+
+### Example usage of ChangeSets in a Tool Function
+
+The following example demonstrates how the Theia Coder agent proposes file modifications using a ChangeSet. This code in embeded in a tool function that Coder provides to the LLM (see also the [full code](https://github.com/eclipse-theia/theia/blob/f4778c2737bb75613f0e1f99da8996bad91f6e17/packages/ai-workspace-agent/src/browser/file-changeset-functions.ts#L60)).
+```typescript
+handler: async (args: string, ctx: ChatRequestModelImpl): Promise<string> => {
+    const { path, content } = JSON.parse(args);
+    const chatSessionId = ctx.session.id;
+    let changeSet = ctx.session.changeSet;
+    if (!changeSet) {
+        changeSet = new ChangeSetImpl('Changes proposed by Coder');
+        ctx.session.setChangeSet(changeSet);
+    }
+    const uri = await this.workspaceFunctionScope.resolveRelativePath(path);
+    let type: 'modify' | 'add' | 'delete' = 'modify';
+    if (content === '') {
+        type = 'delete';
+    }
+    try {
+        await this.fileService.read(uri);
+    } catch (error) {
+        type = 'add';
+    }
```

Comment:
I think using `this.fileService.exists` instead would be more clear

Reply:
I removed the example, but I will update the code